src/plugins/monitoring/src/java/com/reucon/openfire/plugin/archive/impl/MucMamPersistenceManager.java

```diff
@@ -0,0 +1,203 @@
+            if (with != null) {
+                pstmt.setString(++pos, with);
+            }
+            if (xmppResultSet.getAfter() != null) {
+                pstmt.setLong(++pos, xmppResultSet.getAfter());
+            }
+            if (xmppResultSet.getBefore() != null) {
+                pstmt.setLong(++pos, xmppResultSet.getBefore());
+            }
+            rs = pstmt.executeQuery();
+            while (rs.next()) {
+                String senderJID = rs.getString(1);
+                String nickname = rs.getString(2);
+                Date sentDate = new Date(Long.parseLong(rs.getString(3).trim()));
+                String subject = rs.getString(4);
+                String body = rs.getString(5);
+                String stanza = rs.getString(6);
+                long id = rs.getLong(7);
+                if (stanza == null) {
+                    Message message = new Message();
+                    message.setType(Message.Type.groupchat);
+                    message.setSubject(subject);
+                    message.setBody(body);
+                    // Set the sender of the message
+                    if (nickname != null && nickname.trim().length() > 0) {
+                        JID roomJID = room.getRole().getRoleAddress();
+                        // Recreate the sender address based on the nickname and room's JID
+                        message.setFrom(new JID(roomJID.getNode(), roomJID.getDomain(), nickname, true));
+                    }
+                    else {
+                        // Set the room as the sender of the message
+                        message.setFrom(room.getRole().getRoleAddress());
+                    }
+                    stanza = message.toString();
+                }
+                ArchivedMessage archivedMessage = new ArchivedMessage(sentDate, ArchivedMessage.Direction.from, null, null);
+                archivedMessage.setStanza(stanza);
+                archivedMessage.setId(id);
+                msgs.add(archivedMessage);
+            }
+        } catch (SQLException e) {
+            // TODO ???
+        } finally {
+            DbConnectionManager.closeConnection(rs, pstmt, connection);
+        }
+        // TODO - Not great, really should be done by suitable LIMIT stuff.
+        // Would need to reverse ordering in some cases and then reverse results.
+        boolean complete = true;
+        xmppResultSet.setCount(msgs.size());
+        while (msgs.size() > max) {
+            msgs.remove(msgs.size() - 1);
+            complete = false;
+        }
+        xmppResultSet.setComplete(complete);
+        if (msgs.size() > 0) {
+            xmppResultSet.setFirst(msgs.get(0).getId());
+            if (msgs.size() > 1) {
+                xmppResultSet.setLast(msgs.get(msgs.size() - 1).getId());
+            }
+        }
+        return msgs;
+    }
+
+    @Override
+    public Collection<Conversation> getActiveConversations(int conversationTimeout) {
+        return null;
```

Comment:
It can't be called - it's part of the '136 support I'm not attempting to handle for MUC (and not even sure it's possible to handle).